lib/chunky_png/canvas/resampling.rb

```diff
@@ -139,9 +139,25 @@ def resample_bilinear!(new_width, new_height)
       def resample_bilinear(new_width, new_height)
         dup.resample_bilinear!(new_width, new_height)
       end
+
+      def resize_to_fit!(new_width, new_height)
+        shrink_x = new_width.to_f/width
+        shrink_y = new_height.to_f/height
+        if shrink_x < 1.0 && shrink_x < shrink_y
+          self.resize!( new_width, (height * shrink_x).to_i)
+        elsif shrink_y < 1.0
+          self.resize!( (width * shrink_y).to_i, new_height)
+        end
```

Comment:
Is it possible that nether if branch gets triggered?

Reply:
yes and then it leaves the image untouched because it fits in the bounding box